### PR TITLE
Big June 2021 PR 

### PR DESCRIPTION
**Pull request by AndreaCensi**
Mother issue: https://ethidsc.atlassian.net/browse/DTSW-990

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -20,19 +20,19 @@ COPY --from=template /data/config /data/config
 # here, we install the requirements, some requirements come by default
 # you can add more if you need to in requirements.txt
 
-ARG PIP_INDEX_URL
+ARG PIP_INDEX_URL="https://pypi.org/simple"
 ENV PIP_INDEX_URL=${PIP_INDEX_URL}
 RUN echo PIP_INDEX_URL=${PIP_INDEX_URL}
 
-RUN pip install -U "pip>=20.2" pipdeptree
+RUN python3 -m pip install -U   pipdeptree
```
Comment (afdaniele) [inline Dockerfile:27]: format (extra spaces)

File: launchers/run_and_start.sh [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
+dt-exec-BG roscore
+
+dt-exec-BG roslaunch --wait car_interface all.launch veh:="${VEHICLE_NAME}"
+dt-exec-BG roslaunch --wait duckietown_demos lane_following.launch
 sleep 5
 # we put a short sleep in here because rostopic will fail if there's no roscore yet
-rostopic pub /$VEHICLE_NAME/fsm_node/mode duckietown_msgs/FSMState '{header: {}, state: "LANE_FOLLOWING"}'
+
+dt-exec-BG send-fsm-state.sh LANE_FOLLOWING
+#rostopic pub "/${VEHICLE_NAME}/fsm_node/mode" "duckietown_msgs/FSMState" '{header: {}, state: "LANE_FOLLOWING"}' &
```
Comment (afdaniele) [inline launchers/run_and_start.sh]: @AndreaCensi we don't need this anymore?

File: launchers/run_and_start.sh [hunk 3]
```diff
@@ -1,13 +1,25 @@
...
+
+rostopic list
+# foreground
+dt-exec-FG roslaunch --wait agent agent_node.launch || true
+rostopic list
```
Comment (afdaniele) [inline launchers/run_and_start.sh:23]: @AndreaCensi why do we need this?

**Review by afdaniele — CHANGES_REQUESTED** [review]

File: launchers/run_and_start.sh (hunk 2 above)
Comment (liampaull) [reply launchers/run_and_start.sh]: I think it is done by the bash script now

**Pull request merged**